Wikipedia/Code/EditPreviewViewController.swift

```diff
@@ -128,11 +128,54 @@ class EditPreviewViewController: ViewController, WMFPreviewSectionLanguageInfoDe
             showGenericError()
             return
         }
-        messagingController.setup(with: previewWebViewContainer.webView, language: language ?? "en", theme: theme, layoutMargins: articleMargins, areTablesInitiallyExpanded: true)
+        messagingController.setup(with: previewWebViewContainer.webView, language: language ?? "en", theme: theme, layoutMargins: articleMargins, areTablesInitiallyExpanded: true, areEditButtonsHidden: true)
         WMFAlertManager.sharedInstance.showAlert(WMFLocalizedString("wikitext-preview-changes", value: "Retrieving preview of your changes...", comment: "Alert text shown when getting preview of user changes to wikitext"), sticky: false, dismissPreviousAlerts: true, tapCallBack: nil)
         do {
-            let request = try fetcher.mobileHTMLPreviewRequest(articleURL: articleURL, wikitext: wikitext)
+            #if WMF_LOCAL_PAGE_CONTENT_SERVICE
+            // If on local PCS, we need to split this call. On the server, wikitext-to-mobilehtml just puts together two other
+            // calls - wikitext-to-html, and html-to-mobilehtml. Since we have html-to-mobilehtml in local PCS but not the first call, if
+            // we're making PCS edits to mobilehtml we need this code in order to view them. This block splits the call on the local PCS scheme
+            // (similar to what the server does) routing the wikitext-to-html call to production, and html-to-mobilehtml to local PCS.
+            let htmlRequest = try fetcher.wikitextToHTMLRequest(articleURL: articleURL, wikitext: wikitext)
+            let htmlRequestCompletionHandler = { (data: Data?, response: URLResponse?,  error: Error?) in
```

Comment:
I think it'd be nice if we buried the chaining and dataTask details of this into ArticleFetcher. Something like a function signature of 
```
struct SplitResult {
    let html: String
    let response: URLResponse
}
func fetchSplitMobileHTMLCall(articleURL: URL, wikitext: String, completion: (Result<SplitResult, Error>) -> Void)
```

Would give a cleaner call site in the view controller:

```
#if WMF_LOCAL_PAGE_CONTENT_SERVICE
fetcher.fetchSplitMobileHTMLCall(articleURL: articleURL, wikitext: wikitext) { result in
     DispatchQueue.main.async { 
            switch result {
                  case .success(let result):
                         self.previewWebViewContainer.webView.loadHTMLString(result.html, baseURL: result.response?.url)
                  case .failure:
                         self.showGenericError()
             } 
     }
}
#else
...
```

We can still keep the comment on line 135 here explaining it further, but it'd be nice to visually be able to easily skip over the details from the view controller.

Reply:
I did this all with completion blocks - let me know what you think. And please double check the `@escaping` part of it all. I made the `PreviewEditVC`'s completion block use `weak self`, but I always get mildly nervous and want more eyes when I'm doing something that could easily lead to retain cycles.